src/jsonapi-datastore.js

```diff
@@ -100,6 +137,19 @@ class JsonApiDataStore {
    * @param {object} model The model to destroy.
    */
   destroy(model) {
+    var self = this;
+    model._dependents.forEach(function(dependent, depIdx) {
+      self.graph[dependent.type][dependent.id]._removeDependence(model._type, model.id);
+      if (self.graph[dependent.type][dependent.id][dependent.relation].constructor === Array) {
```

Comment:
This probably belong in a `removeModelFromRelationship(type, id, relationshipKey)` method on the `Model`.